src/server/saved_objects/service/lib/schema.js

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export class SavedObjectsSchema {
```

Comment:
I'm not categorically opposed to the name `schema`, but I'm curious if you have a vision for what else might end up in here. I think that namespace agnostic types make sense in here, but I'm wondering if `schema` is overly generic? Don't feel forced to change this, I just want your thoughts on it

Reply:
I'm honestly not sure... I know that long-term we'd like to provide an interface to allow plugins to register their saved object types without specifying their direct Elasticsearch mappings, so this naming was more based around the type of information that we'd like to introspect about the saved objects schema. I struggled with what to call this, so if there's anything more specific that you, or anyone else, has in mind I'm open to alternatives.

Reply:
I don't have anything better 😄